packages/language-server/src/plugins/typescript/language-service.ts

```diff
@@ -123,7 +123,10 @@ async function createLanguageService(
 
 	// Before Astro 1.0, JSX definitions were inside of the language-server instead of inside Astro
 	// TODO: Remove this and astro-jsx.d.ts in types when we consider having support for Astro < 1.0 unnecessary
-	if (astroVersion.major === 0 || astroVersion.full === '1.0.0-beta.0') {
+	if (
+		(astroVersion.major === 0 || astroVersion.full === '1.0.0-beta.0') &&
+		!astroVersion.full.startsWith('0.0.0-rc-') // 1.0.0's RC is considered to be 0.0.0, so we have to check for it
```

Comment:
Unrelated to the PR really, just figured I'd fix that warning at the same time

This fix won't scale into the future of Astro 2.0, but, by then this code should hopefully have been removed 😓